### DISCUSSION TITLE
Programatically start 'password-reset' flow

### DISCUSSION THREAD

**Discussion by sebastian-tschen** [keycloak/keycloak#24996] [Q&A]
We are using a custom flow for the 'password-reset' aka 'forgot-password' flow, where the user is asked to enter a OTP delivered via SMS. The flow works fine, if the user initiates the process via the 'forgot-passwor' link on the login page.
Now we need to initiate this flow programatically for specific users.
We did find the `PUT /admin/realms/{realm}/users/{id}/execute-actions-email
` at the [users API-Endpoint ](https://www.keycloak.org/docs-api/22.0.1/rest-api/index.html#_users), which accepts UPDATE_PASSWORD as action. The user then get's an email to reset his/her password, but the flow differs from the 'reset-password' flow, missing the SMS-OTP step and the user is directly asked to change his/her password.

So we either need a way to programatically trigger the 'reset-password' flow, or change the flow used when the user is aked to change the password via requiredActions.

Any Idea how to solve this?

**Comment by SergeyZubatkin** [comment]
Hi @sebastian-tschen 
We have similar use case, curious whether you were able to solve this issue.
